src/Entries/Entry.php

```diff
@@ -42,15 +44,48 @@ public function toModel()
 
         $data = $this->data();
 
+        $origin = $this->origin();
+        $date = $this->hasDate() ? $this->date() : null;
+
+        if ($blueprint = $this->blueprint()) {
```

Comment:
@ryanmitchell We ran into a few issues with this block regarding fields not properly being detected as inherited, which is why I made a few changes to my PR.
https://github.com/statamic/eloquent-driver/pull/144/files#diff-31fcf8a95fd0dbc8da9e8dc7ae856aa51bd921c50a8059e23df7d411737ab65fR47

Reply:
Can you explain the issues you were seeing?

Reply:
Reverting through CP - I guess through code it would just be unsetting the fields?

Reply:
Got you now, and was able to replicate it.
I've pushed a fix for that.